File: Sessions/Lab7/Code/Models/Quiz.cs
namespace Lab7.Models;

public class Quiz
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = "";
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // ADD THIS LINE: The stable user ID from the token
    public string OwnerId { get; set; } = "";

    public List<Question> Questions { get; set; } = [];
}

File: Sessions/Lab8/Code/src/types/quiz.ts
export type ApiQuiz = {
  id: string; title: string;
  description?: string; ownerId: string;
};

export type Answer   = { id: string; text: string; isCorrect?: boolean; };
export type Question = { id: string; text: string; answers: Answer[]; };
export type Quiz     = { id: string; title: string; questions: Question[]; };